netmaster/objApi/apiController.go

```diff
@@ -0,0 +1,639 @@
+	err := tenant.Write()
+	if err != nil {
+		log.Errorf("Error updating tenant state(%+v). Err: %v", tenant, err)
+		return err
+	}
+
+	return nil
+}
+
+// AppUpdate updates app
+func (ac *APIController) AppUpdate(app, params *contivModel.App) error {
+	log.Infof("Received AppUpdate: %+v, params: %+v", app, params)
+
+	return nil
+}
+
+// AppDelete delete the app
+func (ac *APIController) AppDelete(app *contivModel.App) error {
+	log.Infof("Received AppDelete: %+v", app)
+	return nil
+}
+
+// EndpointGroupCreate creates end point group
+func (ac *APIController) EndpointGroupCreate(endpointGroup *contivModel.EndpointGroup) error {
+	log.Infof("Received EndpointGroupCreate: %+v", endpointGroup)
+	return nil
+}
+
+// EndpointGroupUpdate updates endpoint group
+func (ac *APIController) EndpointGroupUpdate(endpointGroup, params *contivModel.EndpointGroup) error {
+	log.Infof("Received EndpointGroupUpdate: %+v, params: %+v", endpointGroup, params)
+	return nil
+}
+
+// EndpointGroupDelete deletes end point group
+func (ac *APIController) EndpointGroupDelete(endpointGroup *contivModel.EndpointGroup) error {
+	log.Infof("Received EndpointGroupDelete: %+v", endpointGroup)
+	return nil
+}
+
+// NetworkCreate creates network
+func (ac *APIController) NetworkCreate(network *contivModel.Network) error {
+	log.Infof("Received NetworkCreate: %+v", network)
+
+	// Make sure tenant exists
+	if network.TenantName == "" {
+		return errors.New("Invalid tenant name")
+	}
+
+	tenant := contivModel.FindTenant(network.TenantName)
+	if tenant == nil {
+		return errors.New("Tenant not found")
+	}
+
+	// Setup links
+	modeldb.AddLink(&network.Links.Tenant, tenant)
+	modeldb.AddLinkSet(&tenant.LinkSets.Networks, network)
+
+	// Save the tenant too since we added the links
+	err := tenant.Write()
+	if err != nil {
+		log.Errorf("Error updating tenant state(%+v). Err: %v", tenant, err)
+		return err
+	}
+
+	// Get the state driver
+	stateDriver, err := utils.GetStateDriver()
+	if err != nil {
+		return err
+	}
+
+	// Build networ config
+	networkCfg := intent.ConfigNetwork{
+		Name:       network.NetworkName,
+		PktTagType: network.Encap,
+		PktTag:     "",
+		SubnetCIDR: network.Subnet,
+		DefaultGw:  network.DefaultGw,
+	}
+
+	// Create the network
+	err = master.CreateNetwork(networkCfg, stateDriver, network.TenantName)
+	if err != nil {
+		log.Errorf("Error creating network {%+v}. Err: %v", network, err)
+		return err
+	}
+
+	return nil
+}
+
+// NetworkUpdate updates network
+func (ac *APIController) NetworkUpdate(network, params *contivModel.Network) error {
+	log.Infof("Received NetworkUpdate: %+v, params: %+v", network, params)
+	return errors.New("Cant change network parameters after its created")
+}
+
+// NetworkDelete deletes network
+func (ac *APIController) NetworkDelete(network *contivModel.Network) error {
+	log.Infof("Received NetworkDelete: %+v", network)
+
+	// Find the tenant
+	tenant := contivModel.FindTenant(network.TenantName)
```

Comment:
I'm not assuming any orchestrator. Network object always assumes there is a tenant object its referring to. But, i see your point. I'll change it to delete network even it can't find the tenant.